test/e2e/app-dir/actions/app-action.test.ts

```diff
@@ -182,6 +182,20 @@ describe('app-dir action handling', () => {
     }, 'setCookieAndRedirect')
   })
 
+  it('should replace current route when redirecting with type set to replace', async () => {
+    const browser = await next.browser('/header')
+
+    await browser.elementByCss('#setCookieAndRedirectReplace').click()
+    await check(async () => {
+      return (await browser.elementByCss('#redirected').text()) || ''
+    }, 'redirected')
+
+    // Ensure we cannot navigate back
+    const historyLen = await browser.eval('window.history.length')
+    // chromium's about:blank page is the first item in history
+    expect(historyLen).toBe(2)
```

Comment:
I think it'd be nice to also assert the history length before clicking the button so we can verify that it didn't change